### PR TITLE
[msgpack11] Add new port

### PR DESCRIPTION
**Pull request by SanaJahan**
**Describe the pull request**

- What does your PR fix? Fixes issue [#9465 ](https://github.com/microsoft/vcpkg/issues/9465)
  support for a new port: msgpack11 - a tiny MsgPack library for C++11, providing MsgPack parsing and serialization.

- Which triplets are supported/not supported? Have you updated the CI baseline?
 tested in x64-windows-static, osx, linux
- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

### CODE REVIEW

File: ports/msgpack11/CONTROL [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Source: msgpack11
+Version: v0.0.10
+Build-Depends: gtest
+Description: msgpack11 is a tiny MsgPack library for C++11, providing MsgPack parsing and serialization.This library is inspired by json11.The API of msgpack11 is designed to be similar with json11.
+
```
Comment (JackBoosY) [inline ports/msgpack11/CONTROL]: Could you add `Homepage` here?

File: ports/msgpack11/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,26 @@
+include(vcpkg_common_functions)
```
Comment (JackBoosY) [inline ports/msgpack11/portfile.cmake]: We no longer need to include this, please remove this line.

File: ports/msgpack11/msvc.patch [hunk 3]
```diff
@@ -0,0 +1,19 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 9424838..080cefa 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -6,9 +6,12 @@ option(MSGPACK11_BUILD_TESTS "Build unit tests" ON)
+ set(CMAKE_CXX_STANDARD 11)
+ set(CMAKE_CXX_STANDARD_REQUIRED ON)
+ 
+-add_library(msgpack11 msgpack11.cpp)
++add_library(msgpack11 STATIC msgpack11.cpp)
```
Comment (JackBoosY) [inline ports/msgpack11/msvc.patch]: If this port only support static, please add `vcpkg_check_linkage(ONLY_STATIC_LIBRARY)` to the top of `portfile.cmake`.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Comment by JackBoosY** [comment]
LGTM, if this PR is ready, please click the button `Ready for review`.

Thanks.

File: ports/msgpack11/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    	-DMSGPACK11_BUILD_EXAMPLES=OFF
+)
+
+
+vcpkg_install_cmake()
+vcpkg_copy_pdbs()
+
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
+file(COPY ${SOURCE_PATH}/LICENSE.txt DESTINATION ${CURRENT_PACKAGES_DIR}/share/msgpack11)
+file(RENAME ${CURRENT_PACKAGES_DIR}/share/msgpack11/LICENSE.txt ${CURRENT_PACKAGES_DIR}/share/msgpack11/copyright)
```
Comment (PhoebeHui) [inline ports/msgpack11/portfile.cmake]: Could you please update 

 file(COPY ${SOURCE_PATH}/LICENSE.txt DESTINATION ${CURRENT_PACKAGES_DIR}/share/msgpack11)
 file(RENAME ${CURRENT_PACKAGES_DIR}/share/msgpack11/LICENSE.txt ${CURRENT_PACKAGES_DIR}/share/msgpack11/copyright)
to 
file(INSTALL ${SOURCE_PATH}/LICENSE.txt DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)

**Review by PhoebeHui — CHANGES_REQUESTED** [review]

File: ports/msgpack11/portfile.cmake [hunk 5]
```diff
@@ -0,0 +1,26 @@
+vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
+vcpkg_from_github(
+	OUT_SOURCE_PATH SOURCE_PATH
```
Comment (PhoebeHui) [inline ports/msgpack11/portfile.cmake]: Could you update it to align with REPO below? it should be space before it.

File: ports/msgpack11/portfile.cmake [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    REPO ar90n/msgpack11
+    REF v0.0.10
+    SHA512 7b90893f9cdec529789f6e75703f5945c6fc5c946b8708a7a2cb295faf4af111c8cc61265b636f385641031b85181929205be9c5d155f405909445dce85b4ce8
+    HEAD_REF master
+    PATCHES
+    	msvc.patch
+)
+
+vcpkg_configure_cmake(
+	SOURCE_PATH ${SOURCE_PATH}
```
Comment (PhoebeHui) [inline ports/msgpack11/portfile.cmake]: Same as above.

**Review by PhoebeHui — CHANGES_REQUESTED** [review]

File: ports/msgpack11/portfile.cmake (hunk 6 above)
Comment (SanaJahan) [reply ports/msgpack11/portfile.cmake]: made the changes. Let me know if anything is required.

**Review by JackBoosY — APPROVED** [review]

File: ports/msgpack11/CONTROL [hunk 7]
```diff
@@ -0,0 +1,6 @@
+Source: msgpack11
+Version: v0.0.10
```
Comment (PhoebeHui) [inline ports/msgpack11/CONTROL]: The version should be '0.0.10', please remove 'v'.

**Review by PhoebeHui — APPROVED** [review]

**Pull request merged**